src/spec-node/featuresCLI/publishCommandImpl.ts

```diff
@@ -0,0 +1,92 @@
+import { fetchRegistryAuthToken, HEADERS } from '../../spec-configuration/containerFeaturesOCI';
+import { request } from '../../spec-utils/httpRequest';
+import { Log, LogLevel } from '../../spec-utils/log';
+
+const semverCompare = require('semver-compare');
+const semver = require('semver');
+
+interface versions {
+    name: string;
+    tags: string[];
+}
+
+export async function getPublishedVersions(featureId: string, registry: string, namespace: string, output: Log) {
+    try {
+        const url = `https://${registry}/v2/${namespace}/${featureId}/tags/list`;
+        const resource = `${registry}/${namespace}/${featureId}`;
+
+        let authToken = await fetchRegistryAuthToken(output, registry, resource, process.env, 'pull');
+
+        if (!authToken) {
+            output.write(`(!) ERR: Failed to publish feature: ${resource}`, LogLevel.Error);
+            process.exit(1);
+        }
+
+        const headers: HEADERS = {
+            'user-agent': 'devcontainer',
+            'accept': 'application/json',
+            'authorization': `Bearer ${authToken}`
+        };
+
+        const options = {
+            type: 'GET',
+            url: url,
+            headers: headers
+        };
+
+        const response = await request(options);
+        const publishedVersionsResponse: versions = JSON.parse(response.toString());
+
+        return publishedVersionsResponse.tags;
+    } catch (e) {
+        // Publishing for the first time
+        if (e?.message.includes('HTTP 404: Not Found')) {
+            return [];
+        }
+
+        output.write(`(!) ERR: Failed to publish feature: ${e?.message ?? ''} `, LogLevel.Error);
+        process.exit(1);
```

Comment:
This is running in a `for` loop right? Unsure if we should exit here or not.

Reply:
Removed